### ISSUE TITLE
Azure DevOps integration fails with Data Factory deployment

### ISSUE THREAD

**Issue by tlaur** [terraform-providers/terraform-provider-azurerm#3224]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform 0.11.13
AzureRM 1.25.0 (Unreleased)

### Affected Resource(s)

* `azurerm_data_factory`

### Terraform Configuration Files

```hcl

data "azurerm_client_config" "current" {}

resource "azurerm_resource_group" "test_rg" {
  name     = "rg-df-test"
  location = "westeurope"
}

resource "azurerm_data_factory" "test_data_factory" {
  name                = "test-azurerm-df"
  resource_group_name = "rg-df-test"
  location            = "westeurope"

  identity{
    type  = "SystemAssigned"
  }

  vsts_configuration{
    account_name      = "valid_devops_account_name",
    branch_name       = "master",
    project_name      = "testdf",
    repository_name   = "data-factory",
    root_folder       = "/",
    tenant_id         = "${data.azurerm_client_config.current.tenant_id}"
  }
}
```

### Expected Behavior

Data factory deployed with Azure DevOps connection to configured account,project,repo, and branch functioning.

### Actual Behavior

Deployment fails, with error:
azurerm_data_factory.test_data_factory: Can not parse "vsts_configuration.0.tenant_id" as a resource id: Cannot parse Azure ID: parse <tenant-id-here>: invalid URI for request

This is because tenant_id expects a Resource ID, and cannot parse the tenant ID that is passed in the above configuration. However as far as I understand it, the tenant ID is what gets used during manual configuration of the DevOps account and should therefore actually be a valid input. 

Indeed, if the validation is changed to "validate.NoEmptyStrings" as with the other inputs, then the above configuration works and data factory is deployed with the Azure DevOps integration functioning as expected.

```
"tenant_id": {
	Type:         schema.TypeString,
	Required:     true,
	ValidateFunc: **azure.ValidateResourceID**,
},
```
### Steps to Reproduce

1. Compile current azure rm from master
2. `terraform apply`

### Important Factoids

Apologies if this discussion was already had before merging into master, I only had a brief look. VSTS has been rebranded as Azure DevOps, and it might make more sense to use this name in configuration as well, which would be in line with the decision to drop the v2 tag.

* #0000

**Comment by tombuildsstuff** [comment]
hey @tlaur 

Thanks for opening this issue :)

> Deployment fails, with error:
azurerm_data_factory.test_data_factory: Can not parse "vsts_configuration.0.tenant_id" as a resource id: Cannot parse Azure ID: parse : invalid URI for request
> This is because tenant_id expects a Resource ID, and cannot parse the tenant ID that is passed in the above configuration. However as far as I understand it, the tenant ID is what gets used during manual configuration of the DevOps account and should therefore actually be a valid input.

I'd agree this is a bug - this probably should be validating it's a UUID/GUID, rather than a Resource ID. I'm going to ask @mbfrahry to take a look into this.

> Apologies if this discussion was already had before merging into master, I only had a brief look. VSTS has been rebranded as Azure DevOps, and it might make more sense to use this name in configuration as well, which would be in line with the decision to drop the v2 tag.

For the moment since the Azure SDK calls this `VSTS` we've decided to match that - once it's rebranded/renamed we can look to rename this field in a future release - but generally we'd agree with using the new naming when possible - thanks for the feedback either way 👍 

Thanks!

**Comment by tombuildsstuff** [comment]
Fixed via #3228
